Assets/Scripts/Interactables/Dialogue/DialogueInteractable.cs

```diff
@@ -11,29 +11,75 @@ public class DialogueInteractable : Interactable
 
     protected bool isConversing = false;
 
-    private int activeLineIndex = 0;
+    private int activeLineIndex = 0;
+
+
+    private bool autoPlaying = false;
 
-    public override void OnInteract()  
-    {
-        if (!isConversing)
-        {
-            isConversing = true;
+    public float waitLineTime = 2.0f;
 
-            // Freeze player when conversing
-            player.GetComponent<PlayerController>().CanMove = false;
+    protected virtual void Start()
+    {
+        speakerUINPC = Utils.GetRequiredComponentInChildren<SpeakerUI>(this);
```

Comment:
Good style; I'd prefer if Interactable's "player" GameObject was also accessible at all times so that we don't have to keep calling (very expensive) "Get...Component" functions.  I'd assume this would be possible since the player is a singleton?

Not a priority, but might make things easier to conceptualize later on if you think it's worth the refactoring time.

Reply:
Player I don't think is a Singleton (I'd prefer it not to be personally) but we can do `GameManager.Instance.GetPlayerController()` instead so save on that GetComponent overhead!